axelrod/strategies/meta.py

```diff
@@ -533,3 +534,94 @@ def __init__(self):
                 <= 1]
         super().__init__(team=team)
         self.classifier["long_run_time"] = False
+
+class MemoryDecay(MetaPlayer):
+    """
+    A player utilizes the (default) Tit for Tat stretegy for the first (default) 15 turns,
+    at the same time memorizing the opponent's decisions. After the 15 turns have
+    passed, the player calculates a 'net cooperation score' (NCS) for his opponent,
+    weighing decisions to Cooperate as (default) 1, and to Defect as (default)
+    -2. If the opponent's NCS is below 0, the player Defects; otherwise,
+    he Cooperates.
+
+    The player's memories of his opponent's decisions have a random chance to be
+    altered (i.e., a C decision becomes D or vice versa; default probability
+    is 0.03) or deleted (default probability is 0.1).
+
+    It's necessary to specify EXACT name of the starting strategy if changing
+    the default. Possible strategies can be accessed with the .team attribute.
+
+    Name: Memory Decay
+    """
+    name = 'Memory Decay'
+    classifier = {
+        'memory_depth' : float('inf'),
+        'long_run_time' : False,
+        'stochastic' : True,
+        'makes_use_of' : set(),
+        'inspects_source' : False,
+        'manipulates_source' : False,
+        'manipulates_state' : False
+    }
+
+    def __init__(self, p_memory_delete: float = 0.1, p_memory_alter: float = 0.03,
+                 loss_value: float = -2, gain_value: float = 1,
+                 memory: list = None, start_strategy: str = 'Tit For Tat',
```

Comment:
Let's just pass the actual strategy class: `start_strategy: axl.Player = axl.TitForTat`, then the complicated search code can be simplified.

Something like:

```
self.team = [start_strategy()]
```

Reply:
Hey, so, if i change the `__init__` to 

```
    def __init__(self, p_memory_delete: float = 0.1, p_memory_alter: float = 0.03,
                 loss_value: float = -2, gain_value: float = 1,
                 memory: list = None, start_strategy: axelrod.Player = axelrod.TitForTat,
                 start_strategy_duration: int = 15):
```
I get this when trying to import the library:

```
>>> import axelrod
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/denis/Documents/Axelrod/axelrod/__init__.py", line 17, in <module>
    from .strategies import *
  File "/home/denis/Documents/Axelrod/axelrod/strategies/__init__.py", line 9, in <module>
    from .meta import (
  File "/home/denis/Documents/Axelrod/axelrod/strategies/meta.py", line 538, in <module>
    class MemoryDecay(MetaPlayer):
  File "/home/denis/Documents/Axelrod/axelrod/strategies/meta.py", line 569, in MemoryDecay
    memory: list = None, start_strategy: axelrod.Player = axelrod.TitForTat,
NameError: name 'axelrod' is not defined
```

If I add `import axelrod` to the beginning of `meta.py` (which, I suppose, shouldn't be necessary?), I get this:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/denis/Documents/Axelrod/axelrod/__init__.py", line 17, in <module>
    from .strategies import *
  File "/home/denis/Documents/Axelrod/axelrod/strategies/__init__.py", line 9, in <module>
    from .meta import (
  File "/home/denis/Documents/Axelrod/axelrod/strategies/meta.py", line 539, in <module>
    class MemoryDecay(MetaPlayer):
  File "/home/denis/Documents/Axelrod/axelrod/strategies/meta.py", line 570, in MemoryDecay
    memory: list = None, start_strategy: axelrod.Player = axelrod.TitForTat,
AttributeError: module 'axelrod' has no attribute 'TitForTat'
```

Any ideas?

Reply:
Go with:

```
from axelrod.strategies import TitForTat
...
..., axelrod.Player = TitForTat
```

(This is due to the order of imports when loading the library itself.)